src/components/NamespaceList/ManageVisibilityModal.tsx

```diff
@@ -0,0 +1,225 @@
+          model: RoleBindingModel,
+          resource: roleBinding,
+          queryOptions: {
+            ns: namespace.metadata.name,
+          },
+        });
+      } else {
+        // Delete the role binding to make namespace private
+        // Find the existing role binding from the current data
+        const publicRoleBinding = roleBindings?.find(
+          (rb) =>
+            rb.roleRef.name === VIEWER_ROLE_NAME &&
+            rb.subjects?.some(
+              (subject) => subject.kind === 'Group' && subject.name === 'system:authenticated',
+            ),
+        );
+
+        if (publicRoleBinding) {
+          await K8sQueryDeleteResource({
+            model: RoleBindingModel,
+            queryOptions: {
+              name: publicRoleBinding.metadata.name,
+              ns: namespace.metadata.name,
+            },
+          });
+        }
+      }
+
+      onClose();
+    } catch (err) {
+      setError(`Failed to save visibility setting: ${err.message || String(err)}`);
+    } finally {
+      formikHelpers.setSubmitting(false);
+    }
+  };
+
+  // Check if there's a loading error
+  React.useEffect(() => {
+    if (roleBindingsError) {
+      setError(
+        `Failed to load current visibility state: ${roleBindingsError.message || String(roleBindingsError)}`,
+      );
+    }
+  }, [roleBindingsError]);
+
+  return (
+    <Modal
+      {...modalProps}
+      variant={ModalVariant.small}
+      title="Manage visibility"
+      data-testid="manage-visibility-modal"
+      description={
+        <>
+          Manage visibility for a namespace.{' '}
+          <ExternalLink href="https://placeholder-url.com">
+            Learn more about namespace visibility.
+          </ExternalLink>
+        </>
+      }
+    >
+      {error && (
+        <Alert variant={AlertVariant.danger} title="Error" isInline>
+          {error}
+        </Alert>
+      )}
+      <Formik initialValues={initialValues} onSubmit={handleSubmit} enableReinitialize>
+        {({ values, isSubmitting, handleSubmit: formikHandleSubmit }) => {
+          const hasChanges = values.visibility !== currentVisibility;
+
+          return (
+            <>
+              <Form
+                style={{
+                  paddingInline: 'var(--pf-v5-c-modal-box__body--PaddingLeft)',
+                  paddingBlock: 'var(--pf-v5-global--spacer--md) 0',
+                }}
+              >
+                <RadioGroupField
+                  name="visibility"
+                  options={[
+                    {
+                      value: VisibilityOption.PRIVATE,
+                      label: 'Private (Default)',
+                    },
+                    {
+                      value: VisibilityOption.PUBLIC,
+                      label: 'Public',
+                    },
+                  ]}
+                  required={false}
+                  isDisabled={isSubmitting}
+                />
+              </Form>
+              <div
```

Comment:
Please use one of the patternfly layouts, no need to use `div` with inline styles.
Please look at `DeleteResourceModal.tsx` as an example.